languages/v/config.toml

```
@@ -1,7 +1,6 @@
 name = "V"
 grammar = "v"
-path_suffixes = ["v", "vv", "mod"]
-first_line_pattern = '^#![^\\n]+'
+path_suffixes = ["v", "vv", "vsh", "mod"]
```

```suggestion
path_suffixes = ["v", "vv", "vsh", "mod"]
first_line_pattern = '#![^ v\n]*\/(v|env -S v)( .*)?$'
```
The shebang regex shouldn't be deleted as without it, files without an extension wouldn't have syntax highlighting. Please update the PR to use this regex [`#![^ v\n]*\/(v|env -S v)( .*)?$`](https://regex101.com/r/QQUk0H/1).

---

We have to take into account `env` parameters (see `man env`). I added two more counter examples to your regex:

1. `env` with parameters.
2. Path that contains the letter `v`.

See [`^#!.*/(?:v|env(?: .*)? v)(?: |$)`](https://regex101.com/r/H5YGNm/1)

Using `.*` is fine here as the `first_line_pattern` tries to match only the first line. [See](https://github.com/zed-industries/zed/blob/dcf800ff255c98754c2e39034ab458b978c20dee/crates/language/src/language_registry.rs#L654-L681)